lib/src/fsmonitor.rs

```diff
@@ -166,10 +188,18 @@ pub mod watchman {
                 .resolve_root(working_copy_root)
                 .await
                 .map_err(Error::ResolveRootError)?;
-            Ok(Fsmonitor {
+
+            let monitor = Fsmonitor {
                 client,
                 resolved_root,
-            })
+            };
+
+            // Registering the trigger causes an unconditional evaluation of the query, so
+            // test if it is already registered first.
+            if config.register_trigger && !monitor.is_trigger_registered().await? {
+                monitor.register_trigger().await?;
+            }
```

Comment:
Out of curiosity, when will this trigger be deleted?

Should we delete the existing trigger if the config turned off?

Reply:
Done